lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -0,0 +1,366 @@
+#include <glib.h>               // gboolean, GMainLoop, etc.
+#include <libxml/tree.h>        // xmlNode
+
+#include <pacemaker-internal.h>
+
+#include <crm/crm.h>
+#include <crm/cib.h>
+#include <crm/msg_xml.h>
+#include <crm/common/output_internal.h>
+#include <crm/common/xml.h>
+#include <crm/common/iso8601.h>
+#include <crm/common/ipc_controld.h>
+#include <crm/common/ipc_pacemakerd.h>
+#include <crm/common/mainloop.h>
+
+#include <crm/pacemaker/internal.h>
+
+#define DEFAULT_MESSAGE_TIMEOUT_MS 30000
+
+static guint message_timer_id = 0;
+
+static GMainLoop *mainloop = NULL;
+
+static crm_exit_t exit_code = CRM_EX_OK;
```

Comment:
If it's possible, I'd rather define a struct with whatever global info is needed, instantiate it in the top-level functions and pass it around the other functions (callbacks can use it as user data), instead of static globals. It's not a big deal, but that avoids instantiating things when not needed. We use a lot of globals in our libraries currently, but I'd like to move away from that over time, to reduce their overhead. (It's also often harder to debug issues that involve globals.)